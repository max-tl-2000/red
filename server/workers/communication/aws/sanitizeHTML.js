/*
 * Copyright (c) 2022 Reva Technology Inc., all rights reserved.
 * Unauthorized copying of this file, via any medium is strictly prohibited
 * Licensed under the Elastic License 2.0; you may not use this file except
 * in compliance with the Elastic License 2.0.
 */

import sanitizeHtml from 'sanitize-html';

export const sanitize = (html, textFilter, allowedTags = ['img', 'font', 'span']) =>
  sanitizeHtml(html || '', {
    allowedTags: sanitizeHtml.defaults.allowedTags.concat(allowedTags),
    allowedAttributes: {
      ...sanitizeHtml.defaults.allowedAttributes,
      img: ['src'],
      '*': ['href', 'align', 'src', 'alt', 'center', 'bgcolor', 'style', 'size', 'color'],
    },
    allowedSchemesByTag: {
      img: ['data', 'http', 'https'],
    },
    textFilter,
  });
