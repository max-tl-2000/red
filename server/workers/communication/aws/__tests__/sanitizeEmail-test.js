/*
 * Copyright (c) 2022 Reva Technology Inc., all rights reserved.
 * Unauthorized copying of this file, via any medium is strictly prohibited
 * Licensed under the Elastic License 2.0; you may not use this file except
 * in compliance with the Elastic License 2.0.
 */

import { sanitize } from '../sanitizeHTML';
import { emailSizeColor } from '../fixtures/email-size-color';

describe('sanitizeEmail', () => {
  it('should allow size and color attributes in emails', () => {
    const expected = sanitize(emailSizeColor);
    expect(expected).toMatchSnapshot();
  });
});
