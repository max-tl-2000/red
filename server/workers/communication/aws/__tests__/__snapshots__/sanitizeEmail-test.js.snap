// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sanitizeEmail should allow size and color attributes in emails 1`] = `
"<div>
  <font size=\\"6\\" color=\\"#ff9900\\">This is a test that will happen to see if this is working</font>
  <div>
    <br />
  </div>
  <div>Nothing else will happen here</div>
  <div><br /></div>
  <div>
    <b>
      <font color=\\"#0000ff\\">
        <i>This should have a big font</i>
      </font>
    </b> and some color
  </div>
</div>
<div>
  <br />
  <div>On Fri, Aug 11, 2017 at 8:40 PM, Bill Smith <span>&lt;<a href=\\"mailto:parkmerced@royr.local.envmail.reva.tech\\" target=\\"_blank\\">parkmerced@royr.local.envmail.reva.tech</a>&gt;</span> wrote:
  <br />
  <blockquote style=\\"margin:0 0 0 .8ex;border-left:1px #ccc solid;padding-left:1ex\\">So this is a nice<br /><br />
——<br />
Bill Smith<br />
Leasing Consultant<br />
<br />
Parkmerced Apartments<br />
(800) 123-1234<br />
<a href=\\"mailto:bill.smith@parkmerced.com\\">bill.smith@parkmerced.com</a><br />
</blockquote></div><br /></div>"
`;
